crux-llvm/test-data/golden/T784-fail.good

```diff
@@ -0,0 +1,8 @@
+[Crux] Found counterexample for verification goal
+[Crux]   test-data/golden/T784-fail.c:4:3: error: in main
+[Crux]   exit
+[Crux] 
+[Crux] Found counterexample for verification goal
```

Comment:
I don't think we should be seeing this unreachable code error.

Reply:
Humm... that also defies my expectations.  I wonder if this is because "assume after assert" is not the default anymore.

Reply:
Yeah, a quick test confirms that this is related to the `assertThenAssume` option.

We should probably make concretely false assertions abort regardless of that option.

Reply:
Right, it looks like `abort()` is doing things quite differently from how the override for `exit()` now works. This PR is technically only about `abort()`, so would you be OK if I opened a separate issue about your suggested `assertThenAssume` changes?

Reply:
I've opened https://github.com/GaloisInc/crucible/issues/787 for this.